keps/sig-network/3698-multi-network/README.md

```diff
@@ -0,0 +1,1114 @@
+validations, inside the  scheduler, that require queries for other objects
+(e.g. PodNetwork). These errors will be presented as Pod Events, and the Pod
+will be kept in Pending state until the issues are resolved. We will do the
+following validation:
+* Referenced PodNetwork or PodNetworkAttachment is present
+* Referenced PodNetwork or PodNetworkAttachment is Ready
+* Referenced PodNetworkAttachment, that is referencing a PodNetwork, follows the
+rule of: single Pod references a given PodNetwork only 1 time
+
+#### Auto-population
+When networks field is not set, and hostNetwork is not set, we will
+auto-populate this field with following values:
+```yaml
+networks:
+- podNetworkName: default
+```
+
+This is to ensure backward compatibility for clusters that will not use
+PodNetwork explicitly.
+
+#### Status
+All the IP addresses for attached PodNetworks will be present in the
+Pod.PodStatus.PodIPs list. To properly identify which IP belongs to what
+PodNetwork, we will expand the PodIP struct to include the name of PodNetwork
+the specific IP belongs to. This list will allow only 1 IP address per family
+(v4, v6) per PodNetwork. The Pod.PodStatus.PodIP behavior will not change.
+Proposed changes below:
+```go
+// IP address information for entries in the (plural) PodIPs field.
+// Each entry includes:
+//
+//      IP: An IP address allocated to the pod.
+//      PodNetworkName: Name of the PodNetwork the IP belongs to.
+//      InterfaceName: Name of the network interface inside the Pod.
+type PodIP struct {
+        // ip is an IP address (IPv4 or IPv6) assigned to the pod
+        IP string `json:"ip,omitempty" protobuf:"bytes,1,opt,name=ip"`
+
+        // PodNetworkName is name of the PodNetwork the IP belongs to
+        //
+        // +optional
+        PodNetworkName string `json:"podNetwork"`
+
+        // InterfaceName is name of the network interface used for this attachment
+        //
+        // +optional
+        InterfaceName string `json:"interfaceName",omitempty`
+}
+```
+Example:
+```yaml
+kind: Pod
+metadata:
+  name: pod1
+  namespace: default
+spec:
+[...]
+  networks:
+  - podNetwork: default
+    interfaceName: eth0
+    isDefaultGW4: true
+  - podNetwork: dataplane1
+    interfaceName: net1
+  - podNetworkAttachmentName: my-interface-fast
+    interfaceName: net2
+status:
+[...]
+  podIP: 192.168.5.54
+  podIPs:
+  - ip: 192.168.5.54
+    podNetwork: default
+    interfaceName: eth0
+  - ip: 10.0.0.20
+    podNetwork: dataplane1
+    interfaceName: net1
+  - ip: 2011::233
+    podNetwork: my-interface-fast
+    interfaceName: net2
+```
+The above status is expected to be populated by kubelet, but this can only happen
+after CRI provides support for the new Pod API. Because of that, initially
+kubelet will behave as it does today, without updating the additional fields.
+Until CRI catches up, the PodNetwork providers will be able to update that field
+on their own.
+
```

Comment:
This is an API break. Please see https://github.com/kubernetes/kubernetes/pull/123112#issuecomment-1925957930. IMO this makes perfect sense, but as an API break a way forward _must_ be investigated and described. @danwinship

Reply:
makes sense. 

IMHO, option 2 - `pod.status.networks` -  is preferable (even w/ its downside).

Reply:
updated